items:
- uid: System.Threading.ThreadStateException
  commentId: T:System.Threading.ThreadStateException
  id: ThreadStateException
  parent: System.Threading
  children:
  - System.Threading.ThreadStateException.#ctor
  - System.Threading.ThreadStateException.#ctor(System.String)
  - System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  - vb
  name: ThreadStateException
  fullName: System.Threading.ThreadStateException
  type: Class
  source:
    remote:
      path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: ThreadStateException
    path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
    startLine: 50
  assemblies:
  - System.Threading.Thread
  namespace: System.Threading
  summary: "\n<p>The exception that is thrown when a <xref href=\"System.Threading.Thread\" data-throw-if-not-resolved=\"false\"></xref> is in an invalid <xref href=\"System.Threading.Thread.ThreadState\" data-throw-if-not-resolved=\"false\"></xref> for the method call.</p>\n"
  remarks: "\n<p>Once a thread is created, it is in at least one of the <xref href=\"System.Threading.ThreadState\" data-throw-if-not-resolved=\"false\"></xref> states until it terminates. <strong>ThreadStateException</strong> is thrown by methods that cannot perform the requested operation due to the current state of a thread. For example, trying to restart an aborted thread by calling <xref href=\"System.Threading.Thread.Start\" data-throw-if-not-resolved=\"false\"></xref> on a thread that has terminated throws a <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>\n<xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref> uses the HRESULT COR_E_THREADSTATE, which has the value 0x80131520.</p>\n<p>For a list of initial property values for an instance of <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref>, see the <xref href=\"System.Threading.ThreadStateException.#ctor\" data-throw-if-not-resolved=\"false\"></xref> constructors.</p>\n"
  example:
  - "\n<p>The following example demonstrates an error that causes the system to throw a <strong>ThreadStateException</strong>.</p>ThreadStEx#1<p>This code produces the following output: </p><pre><code> In main. Attempting to restart myThread.\nWorking thread...\nCaught: Thread is running or terminated. Cannot restart.</code></pre>"
  syntax:
    content: 'public class ThreadStateException : Exception'
    content.vb: >-
      Public Class ThreadStateException
          Inherits Exception
  seealso:
  - type: System.Threading.Thread
    commentId: T:System.Threading.Thread
  - type: System.Threading.ThreadState
    commentId: T:System.Threading.ThreadState
  - type: System.Exception
    commentId: T:System.Exception
  inheritance:
  - System.Object
  modifiers.csharp:
  - public
  - class
  modifiers.vb:
  - Public
  - Class
- uid: System.Threading.ThreadStateException.#ctor
  commentId: M:System.Threading.ThreadStateException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  - vb
  name: ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
  type: Constructor
  source:
    remote:
      path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
    startLine: 53
  assemblies:
  - System.Threading.Thread
  namespace: System.Threading
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref> class with default properties.</p>\n"
  remarks: "\n<p>This constructor initializes the <strong>Message</strong> property of the new instance to a system-supplied message that describes the error, such as \"The requested operation cannot be performed on the thread due to its current state.\" This message takes into account the current system culture.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The localized error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public ThreadStateException()
    content.vb: Public Sub New
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  commentId: M:System.Threading.ThreadStateException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  - vb
  name: ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(System.String)
  type: Constructor
  source:
    remote:
      path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
    startLine: 54
  assemblies:
  - System.Threading.Thread
  namespace: System.Threading
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message.</p>\n"
  remarks: "\n<p>This constructor initializes the <strong>Message</strong> property of the new instance using the <em>message</em> parameter.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>\n<strong>null</strong>. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public ThreadStateException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  commentId: M:System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  - vb
  name: ThreadStateException(String, Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(System.String, System.Exception)
  type: Constructor
  source:
    remote:
      path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
      branch: master
      repo: https://github.com/dotnet/corefx.git
    id: .ctor
    path: src/System.Threading.Thread/ref/System.Threading.Thread.cs
    startLine: 55
  assemblies:
  - System.Threading.Thread
  namespace: System.Threading
  summary: "\n<p>Initializes a new instance of the <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.</p>\n"
  remarks: "\n<p>An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property. The <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property returns the same value that is passed into the constructor, or <strong>null</strong> if the <xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref> property does not supply the inner exception value to the constructor.</p>\n<p>The following table shows the initial property values for an instance of <xref href=\"System.Threading.ThreadStateException\" data-throw-if-not-resolved=\"false\"></xref>.</p>\n<p>Property </p>\n<p>Value </p>\n<table><tbody><tr><td>\n<p>\n<xref href=\"System.Exception.InnerException\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The inner exception reference. </p>\n</td></tr><tr><td>\n<p>\n<xref href=\"System.Exception.Message\" data-throw-if-not-resolved=\"false\"></xref>\n</p>\n</td><td>\n<p>The error message string. </p>\n</td></tr></tbody></table>\n"
  syntax:
    content: public ThreadStateException(string message, Exception innerException)
    content.vb: Public Sub New(message As String, innerException As Exception)
    parameters:
    - id: message
      type: System.String
      description: "\n<p>The error message that explains the reason for the exception. </p>\n"
    - id: innerException
      type: System.Exception
      description: "\n<p>The exception that is the cause of the current exception. If the <em>innerException</em> parameter is not <strong>null</strong>, the current exception is raised in a <strong>catch</strong> block that handles the inner exception. </p>\n"
  seealso:
  - type: System.Exception
    commentId: T:System.Exception
  modifiers.csharp:
  - public
  modifiers.vb:
  - Public
references:
- uid: System.Threading
  isExternal: false
  name: System.Threading
  fullName: System.Threading
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
